src/field_theory/finite/galois_field.lean

```diff
@@ -0,0 +1,146 @@
+/-
+Copyright (c) 2021 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Aaron Anderson, Alex J. Best, Johan Commelin, Eric Rodriguez, Ruben Van de Velde
+-/
+
+import algebra.char_p.algebra
+import field_theory.finite.basic
+import field_theory.separable
+import linear_algebra.finite_dimensional
+
+/-!
+# Galois fields
+
+If `p` is a prime number, and `n` a natural number,
+then `galois_field p n` is defined as the splitting field of `X^(p^n) - X` over `zmod p`.
+It is a finite field with `p ^ n` elements.
+
+## Main definition
+
+* `galois_field p n` is a field with `p ^ n` elements
+
+-/
+
+noncomputable theory
+
+open polynomial
+
+lemma galois_poly_separable {K : Type*} [field K] (p q : ℕ) [char_p K p] (h : p ∣ q) :
+  separable (X ^ q - X : polynomial K) :=
+begin
+  use [1, (X ^ q - X - 1)],
+  rw [← char_p.cast_eq_zero_iff (polynomial K) p] at h,
+  rw [derivative_sub, derivative_pow, derivative_X, h],
+  ring,
+end
+
+section splitting_field_facts
+
+variables {K : Type*} {L : Type*} [field K] [field L] {f : polynomial K}
+
+-- linter complains, but removing causes `instance : fintype (galois_field p n)` below to fail
```

Comment:
I don't know how to fix this - should otherwise be good for another round of review

Reply:
I just changed the proof below, maybe there is a smarter way but this seems safe and works to me

Reply:
we can also make `galois_field` reducible (or at least temporarily), which also fixes it; i said this [a while back](https://github.com/leanprover-community/mathlib/pull/8692#discussion_r689148799)

Reply:
@ericrbg yeah I think these two solutions are basically equivalent, I just feel like if you set it reducible even locally you might accidentally rely on reducibility too much? Not a big difference anyway I don't think